Extend roadmap: cookies (#423)
w3c/webdriver-bidi: roadmap.md

- [x] Some items from the previous milestones
- [x] [Printing to PDF as Base64-encoded string](https://w3c.github.io/webdriver-bidi/#command-browsingContext-print)

### Cookies

> This scenario loads a web page and gets and sets its cookies. In spec terms, this involves:

Reviewer: ```suggestion
This scenario loads a web page, reads the cookies set by the server and the page, updates and deletes some of the cookies and triggers an action to verify that modified cookies are processed correctly. In spec terms, this involves:
```

- [x] Some items from the previous milestones

> - [ ] Get cookies
> - [ ] Set cookies
> - [ ] Delete cookies

Reviewer 2: ```suggestion
- [ ] Getting cookies
- [ ] Setting cookies
- [ ] Deleting cookies
```

> This scenario loads a web page, reads the cookies set by the server and the page, updates and deletes some of the cookies, and triggers an action to verify that modified cookies are processed correctly. In spec terms, this involves:

Reviewer: ```suggestion
This scenario starts the browser, reads cookies available in the browser's storage, updates and deletes some of the cookies, and navigates to a page to verify that cookies are processed correctly. In spec terms, this involves:
```

> This scenario loads a web page, reads the cookies set by the server and the page, updates and deletes some of the cookies, and triggers an action to verify that modified cookies are processed correctly. In spec terms, this involves:

Reviewer 3: Maybe something like
```suggestion
This scenario sets cookies that will be used when loading a web page, and its subresources, loads that page, and verifies that the correct cookies were sent with the requests. It also allows inspecting the cookies set by the responses, and clearing cookies to reset state. In spec terms, this involves:
```
although strictly that phrasing only requires _write_ access to the cookie jar when the page hasn't yet been loaded.

> This scenario sets cookies that will be used when loading a web page, and its subresources, loads that page, and verifies that the correct cookies were sent with the requests. It also allows inspecting the cookies set by the responses, and clearing cookies to reset state. In spec terms, this involves:

Reviewer 2: ```suggestion
This scenario sets cookies to be used when loading a web page and its subresources, loads that page, and verifies that the correct cookies were sent with the requests. It also inspects the cookies set by the responses, and clears all cookies to reset the state. In spec terms, this involves:
```

### Cookies

> This scenario loads a web page, reads the cookies set by the server and the page, updates and deletes some of the cookies, and triggers an action to verify that modified cookies are processed correctly. In spec terms, this involves:

Reviewer 3: This doesn't cover cookies for pages that aren't loaded. My understanding is that the need to load a page before interacting with its cookies is considered a significant limitation for WebDriver. It's also the source of much of the complexity here (since having a page loaded in a specific context answers a lot of the questions around keying, given that a particular browsing context will have a particular storage key).

Reviewer 2: Is this addressed by the edit that happened after this comment was posted?
